.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,32 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ main ]
```

Comment:
It looks like this workflow runs independently of any tests that run. I think you should add itas a job to the build and test workflow, and you can configure it to run after the tests run so that we only push images that pass tests.

Reply:
makes sense. i'll do that